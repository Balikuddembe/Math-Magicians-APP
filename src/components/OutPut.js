import PropTypes from 'prop-types';

function OutPut({ state }) {
  const { total, next, operation } = state;
  return (
    <div className="output">
      {`${total || ''} ${operation || ''} ${next || ''}`}
    </div>
  );
}

OutPut.propTypes = {
  state: PropTypes.func.isRequired,
};
export default OutPut;
