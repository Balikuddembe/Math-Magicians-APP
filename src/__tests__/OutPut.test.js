import React from 'react';
import { render } from '@testing-library/react';
import OutPut from '../components/OutPut';

describe('OutPut component', () => {
  test('renders correctly', () => {
    const state = {
      total: '10',
      next: '4',
      operation: '+',
    };
    const tree = render(
      <OutPut state={state} />,
    );
    expect(tree.firstChild).toMatchSnapshot();
  });
});
