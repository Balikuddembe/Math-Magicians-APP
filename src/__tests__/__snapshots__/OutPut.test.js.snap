// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OutPut component renders correctly 1`] = `undefined`;
